Add blog post for rc2 release (#4817)
FerretDB/FerretDB: website/blog/2025-02-25-new-ferretdb-v2-rc2-release-with-debian-packages-and-more-indexing-support.md

---
slug: new-ferretdb-v2-rc2-release-with-debian-packages-and-more-indexing-support
title: FerretDB releases v2.0.0-rc.2 with new Debian packages and more indexing support
authors: [alex]
description: >
  We have released FerretDB v2.0.0-rc.2, with an embeddable package, new Debian packages, improved stats and logging, and more indexing support.
image: /img/blog/ferretdb-v2-rc2.jpg
tags: [release]
---

![FerretDB v2.0.0-rc.2](/img/blog/ferretdb-v2-rc2.jpg)

> [FerretDB](https://www.ferretdb.com/) v2 keeps getting better!

Reviewer: Is that link really necessary in our blog?

> Based on your feedback, we are making improvements to FerretDB v2 with an embeddable package, new Debian packages, improved logging, and more indexing support.

Reviewer: ```suggestion
Based on your feedback, we are making improvements with an embeddable Go package, new Debian packages, improved logging, and better indexing support.
```

Support can't be "more". And it is unclear what an "embeddable package" is. Is it embeddable into PostgreSQL? No.

The same in the title and description

Reviewer: > Fashander marked this conversation as resolved.

https://github.com/FerretDB/FerretDB/blob/d170db2b0798b3035849805f608148b5dfb07124/CONTRIBUTING.md?plain=1#L379

The part about the title and description was ignored.

@Fashander, please focus and review all my comments.

<!--truncate-->

> After [we released FerretDB v2](https://blog.ferretdb.io/ferretdb-releases-v2-faster-more-compatible-mongodb-alternative/), we received lots of amazing feedbacks and shoutouts from the community.

Reviewer: @Fashander Please use some grammar-checking tools. "Feeback" is uncountable.

> After [we released FerretDB v2](https://blog.ferretdb.io/ferretdb-releases-v2-faster-more-compatible-mongodb-alternative/), we received lots of amazing feedbacks and shoutouts from the community.

Reviewer: We did not release _FerretDB v2_

Reviewer: That was ignored…

So many of our users have been impressed with its overall performance improvements and range of compatible features, including vector search, full text search, replication, aggregation pipeline support, time series capabilities, among others.

We've just released FerretDB v2.0.0-rc.2, with new features and improvements that make it even better, and we're excited to share them with you!

## Docker image tags update

We've made some changes to our Docker image tags.
The `latest` tag now points to v2, which means if you're pulling directly without specifying a tag (e.g. `ghcr.io/ferretdb/ferretdb`), you're getting FerretDB v2 by default.
If you're running FerretDB in production, we strongly recommend explicitly specifying the full version tag (e.g. `ghcr.io/ferretdb/ferretdb:2.0.0-rc.2`) to ensure consistency across deployments.

> ## Embeddable package
> 
> FerretDB is now even easier to integrate into other applications with the new embeddable package.

Reviewer: ```suggestion
## Embeddable Go package

FerretDB is now even easier to integrate into other applications with the new embeddable Go package.
```

Otherwise, it would be safe to assume it is a C package, but it is not.

This means you can run FerretDB within your own Go applications, giving you more flexibility in deployment and usage.

## Debian packages for DocumentDB

We now provide `.deb` packages for users who want to install FerretDB with the DocumentDB PostgreSQL extension on Debian-based systems.
This makes installation and upgrades simpler for Debian and Ubuntu users.

> You can now install it from our repository; you can also check out the installation guide in our documentation.

Reviewer: From _what_ our repository?

> ## Support for TTL index and `reIndex`

Reviewer: ```suggestion
## Support for TTL indexes and `reIndex` command
```

It's not like we support only one TTL index…

Reviewer: It is still TTL index…

Indexes are essential parts of any database, and based on your feedback, we now have full support for TTL indexes and the `reIndex` command.
You can try it out as shown below:

TTL indexes now function correctly; Expired documents will now automatically be removed as expected.

```sh
db.collection.createIndex( { "createdAt": 1 }, { expireAfterSeconds: 3600 } )
```

> Similarly, the `reIndex` command has been fixed, so you can rebuild indexes without issues.

Reviewer: It was implemented, not fixed. Nothing "similarly" about it. We should explicitly recommend that users use it to rebuild indexes, as there were problems with all indexes, not only TTL ones.

```sh
db.collection.reIndex()
```

## New `dbStats` command

`dbStats` is a crucial command for many UI applications and monitoring tools.
It provides important information about database size, storage efficiency, and more.
With the new release, you can now get better insights into your database with the `dbStats` command.

```sh

> db.stats()

Reviewer: Yet the example shows a different command.
I think we really should start using `runCommand` in all examples and then mention `mongosh` wrappers for them.

```

## Mongo-compatible logging format

> Logs now follow the MongoDB-style format, making it easier for users and tools to parse and analyze them.

Reviewer: They don't by default.

## Thank you!

> With the release of FerretDB v2, we are enabling more applications to run their MongoDB workloads in open source, using all the familiar features and commands.

Reviewer: There is no "release of FerretDB v2" – there is only release candidate

> With the release of FerretDB v2, we are enabling more applications to run their MongoDB workloads in open source, using all the familiar features and commands.

Reviewer: "to run their MongoDB workloads in open source" implies that workloads should be open source now. I guess you mean "on".

We are grateful for all the feedback, suggestions, and contributions from the community, and we look forward to hearing more from you as we continue to improve FerretDB.

If you're still on v1, you're missing out on all these new features and improvements.

> FerretDB v2 is the most feature-complete open source alternative to MongoDB – [try it out here](https://github.com/FerretDB/FerretDB/releases)!

Reviewer: "here" is weird – you can't try it out on that page

> If you have any questions about FerretDB, please feel free to [reach out on any of our channels here](https://docs.ferretdb.io/#community).

Reviewer: "listed here" or something
